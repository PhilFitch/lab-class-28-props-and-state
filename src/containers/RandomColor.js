import React, { Component } from 'react';
import ColorChanger from '../components/colors/ColorChanger';

export default class RandomColor extends Component {
  state = {
    backgroundColor: 'black'
  };

  chooseColor(colors) {
    let idx = Math.floor((Math.random()) * (colors.length));
    let color = colors[idx];
    console.log(idx);
    return {
      backgroundColor: color
    };
  }
  
  componentDidMount() {
    const colors = ['red', 'green', 'black', 'yellow', 'purple', 'blue', 'orange', 'violet', 'lavender', 'brown'];
    this.changeColor(colors, this.chooseColor);
  }

  changeColor(colors, chooseColor) {
    setInterval(() => this.setState(chooseColor(colors)), 1000);
  }
  
  render() {
    
    return (
      <ColorChanger backgroundColor={this.state.backgroundColor} />
    );
  }
}
