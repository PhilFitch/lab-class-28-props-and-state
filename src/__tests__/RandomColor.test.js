import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from '../containers/RandomColor';

describe('RandomColor component', () => {
  it('renders RandomColor', () => {
    const wrapper = shallow(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
