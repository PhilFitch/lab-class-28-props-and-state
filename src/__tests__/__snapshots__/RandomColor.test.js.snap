// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomColor component renders RandomColor 1`] = `
<ColorChanger
  backgroundColor="black"
/>
`;
